packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx

```diff
@@ -0,0 +1,515 @@
+            { label: "25", value: "25" },
+            { label: "50", value: "50" },
+            { label: "All", value: String(this.totalItems) },
+          ]
+        : [
+            { label: "25", value: "25" },
+            { label: "100", value: "100" },
+            { label: "1000", value: "1000" },
+            { label: "All", value: String(this.totalItems) },
+          ];
+  };
+
+  private setGoToPageInputStyles = () => {
+    const textField = this.el.shadowRoot?.querySelector(".go-to-page-input");
+    if (textField !== undefined) {
+      const input = textField?.shadowRoot?.querySelector("input");
+      if (input !== undefined) {
+        input.style.textAlign = "center";
+        input.style.padding = "0";
+      }
+    }
+  };
+
+  private setNumberPages = () => {
+    this.totalPages = Math.ceil(this.totalItems / Number(this.itemsPerPage));
+  };
+
+  private setUpperBound = () => {
+    this.upperBound = Math.min(
+      this.lowerBound + Number(this.itemsPerPage) - 1,
+      this.totalItems
+    );
+  };
+
+  private trimItemsPerPageOptions = () => {
+    this.itemsPerPageOptions = this.itemsPerPageOptions.slice(0, 3);
+    this.itemsPerPageOptions.push({
+      label: "All",
+      value: String(this.totalItems),
+    });
+
+    for (let i = 0; i < this.itemsPerPageOptions.length - 1; i++) {
+      if (this.totalItems <= Number(this.itemsPerPageOptions[i].value)) {
+        this.itemsPerPageOptions.splice(
+          i,
+          this.itemsPerPageOptions.length - (i + 1)
+        );
+      }
+    }
+  };
+
+  render() {
+    const {
+      appearance,
+      alignment,
+      itemsPerPageOptions,
+      paginationControl,
+      paginationType,
+      showItemsPerPage,
+      showItemsPerPageControl,
+      showGoToPageControl,
+    } = this;
+
+    return (
+      <div
+        class={{
+          ["pagination-bar"]: true,
+          [`pagination-bar-${alignment}`]: true,
+        }}
+        ref={(el) => (this.paginationBarEl = el)}
+      >
+        {showItemsPerPage || showItemsPerPageControl ? (
+          <div class="item-controls">
+            {showItemsPerPageControl ? (
+              <div class="items-per-page-holder">
+                <ic-typography
+                  class={{
+                    [`pagination-text-${appearance}`]: true,
+                    ["items-per-page-control-label"]: true,
+                  }}
+                  variant="label"
+                >
+                  {this.itemLabel}s per {this.pageLabel.toLowerCase()}
+                </ic-typography>
+                <ic-select
+                  small
+                  label="items-per-page-input"
+                  class="items-per-page-input"
+                  hideLabel
+                  options={itemsPerPageOptions}
+                  value={this.itemsPerPage}
+                  onIcChange={() => this.changeItemsPerPage()}
+                ></ic-select>
+              </div>
+            ) : null}
```

Comment:
No need to return `null` if nothing is going to be rendered, just use `{showItemsPerPageControl && (<element>)`. There are a couple instances where this can be applied in pagination and data-table